### ISSUE TITLE
Paper Mario Sticker Star - Major graphical issues

### ISSUE THREAD

**Issue by 0x0000null** [citra-emu/citra#2344]
All particles in Paper Mario: Sticker Star seem to be incorrectly rendered by Citra.

> macOS Sierra version 10.12.1  
> Processor: 3.3 GHz Intel Core i5  
> Graphics: AMD Radeon R9 M290 2048 MB  
> Citra: `master-d30d6f8`

<img width="1200" alt="1" src="https://cloud.githubusercontent.com/assets/16504771/21284820/2e7b5c26-c3dc-11e6-9ed8-20dd4cf444ba.png">|<img width="1200" alt="2" src="https://cloud.githubusercontent.com/assets/16504771/21284821/2e7ccf52-c3dc-11e6-8a4d-00a5a1202ad1.png">
:--:|:--:
<img width="1200" alt="3" src="https://cloud.githubusercontent.com/assets/16504771/21284822/2e7de162-c3dc-11e6-90f7-2cf6348acf64.png">|<img width="1200" alt="4" src="https://cloud.githubusercontent.com/assets/16504771/21284823/2e7e3504-c3dc-11e6-9903-909e8784bdd7.png">

**On hardware:** (apologies for potato quality)|<img alt="5" src="https://cloud.githubusercontent.com/assets/16504771/21284824/2eac290a-c3dc-11e6-9460-5f68ebfc1879.jpg">
---:|:--:

Log:

```
[   6.006769] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:495: GL_VERSION: 4.1 ATI-1.46.22
[   6.006798] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:496: GL_VENDOR: ATI Technologies Inc.
[   6.006803] Render.OpenGL <Info> video_core/renderer_opengl/renderer_opengl.cpp:Init:497: GL_RENDERER: AMD Radeon R9 M290 OpenGL Engine
[   6.041172] Loader <Info> core/hle/kernel/process.cpp:ParseKernelCaps:99: ExHeader kernel version: 2.33
[   6.451596] Service.SRV <Warning> core/hle/service/srv.cpp:RegisterClient:40: (STUBBED) called
[   6.451950] Service.SRV <Warning> core/hle/service/srv.cpp:EnableNotification:64: (STUBBED) called
[   6.453807] Service.APT <Warning> core/hle/service/apt/apt.cpp:GetLockHandle:120: (STUBBED) called handle=0x00038009 applet_attributes=0x00000000
[   6.456494] Service.NDM <Warning> core/hle/service/ndm/ndm.cpp:OverrideDefaultDaemons:201: (STUBBED) default_daemon_bit_mask=0x0000000F 
[   6.456602] Service.NDM <Warning> core/hle/service/ndm/ndm.cpp:SuspendDaemons:87: (STUBBED) daemon_bit_mask=0x00000009 
[   6.457959] Service.FS <Warning> core/hle/service/fs/fs_user.cpp:InitializeWithSdkVersion:798: (STUBBED) called, version: 0x040200C8
[   6.463656] Service.APT <Warning> core/hle/service/apt/apt.cpp:SetAppCpuTimeLimit:332: (STUBBED) called cpu_percent=30, value=1
[   6.464310] Service.APT <Warning> core/hle/service/apt/apt.cpp:Enable:128: (STUBBED) called attributes=0x00000000
[   6.465744] Service.APT <Warning> core/hle/service/apt/apt.cpp:GlanceParameter:247: called app_id=0x00000300, buffer_size=0x00001000
[   6.466585] Service.APT <Warning> core/hle/service/apt/apt.cpp:NotifyToWait:102: (STUBBED) app_id=768
[   6.467924] Service.APT <Warning> core/hle/service/apt/apt.cpp:ReceiveParameter:224: called app_id=0x00000300, buffer_size=0x00001000
[   6.469802] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:318: (STUBBED) called command=0x00000007, buffer1_size=0x00000004, buffer2_size=0x00000001, buffer1_addr=0x0FFFFF08, buffer2_addr=0x0FFFFED4
[   6.470128] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:318: (STUBBED) called command=0x00000004, buffer1_size=0x00000001, buffer2_size=0x00000001, buffer1_addr=0x0FFFFED0, buffer2_addr=0x0FFFFED4
[   6.481844] Service.HID <Warning> core/hle/service/hid/hid.cpp:GetGyroscopeLowCalibrateParam:267: (STUBBED) called
[   6.502982] Service.GSP <Warning> core/hle/service/gsp_gpu.cpp:AcquireRight:690: called
[   6.503746] Service.GSP <Warning> core/hle/service/gsp_gpu.cpp:RegisterInterruptRelayQueue:384: called, flags=0x00000001
[   6.508138] HW.GPU <Critical> core/hw/gpu.cpp:MemoryFill:96: invalid start address 0x00000000
[   6.508187] HW.GPU <Critical> core/hw/gpu.cpp:MemoryFill:96: invalid start address 0x00000000
[   6.550165] HW.GPU <Critical> core/hw/gpu.cpp:MemoryFill:101: invalid end address 0x10B00000
[   6.601637] HW.Memory <Error> core/memory.cpp:VirtualToPhysicalAddress:693: Unknown virtual address @ 0x0043D408
[   6.601766] HW.Memory <Error> core/memory.cpp:VirtualToPhysicalAddress:693: Unknown virtual address @ 0x0043D414
[   6.604280] HW.Memory <Error> core/memory.cpp:VirtualToPhysicalAddress:693: Unknown virtual address @ 0x003EB020
[   6.604963] HW.Memory <Error> core/memory.cpp:VirtualToPhysicalAddress:693: Unknown virtual address @ 0x003EB260
[   6.697186] Service.DSP <Info> core/hle/service/dsp_dsp.cpp:LoadComponent:142: Firmware hash: 0x59c64be916c7fd28
[   6.697216] Service.DSP <Info> core/hle/service/dsp_dsp.cpp:LoadComponent:146: Structures hash: 0x93e48663943cad39
[   6.697221] Service.DSP <Warning> core/hle/service/dsp_dsp.cpp:LoadComponent:150: (STUBBED) called size=0xC20A, prog_mask=0x505300FF, data_mask=0x000D00FF, buffer=0x00441278
[   6.697784] Service.DSP <Info> core/hle/service/dsp_dsp.cpp:RegisterInterruptEvents:241: Registered type=2, pipe=2, event_handle=0x000E002B
[   6.698044] Service.DSP <Warning> core/hle/service/dsp_dsp.cpp:GetSemaphoreEventHandle:167: (STUBBED) called
[   6.698283] Service.DSP <Warning> core/hle/service/dsp_dsp.cpp:SetSemaphoreMask:446: (STUBBED) called mask=0x00002000
[   6.698620] Audio.DSP <Info> audio_core/hle/pipe.cpp:PipeWrite:134: Application has requested initialization of DSP hardware
[   6.698825] Service.DSP <Warning> core/hle/service/dsp_dsp.cpp:SetSemaphore:264: (STUBBED) called
[   6.700936] Service.DSP <Warning> core/hle/service/dsp_dsp.cpp:SetSemaphore:264: (STUBBED) called
[   6.711951] Kernel.SVC <Warning> core/hle/svc.cpp:CreateThread:565: Newly created thread is allowed to be run in the SysCore, unimplemented.
[   6.911292] Service.APT <Warning> core/hle/service/apt/apt.cpp:AppletUtility:318: (STUBBED) called command=0x00000004, buffer1_size=0x00000001, buffer2_size=0x00000001, buffer1_addr=0x0FFFFFA8, buffer2_addr=0x0FFFFFAC
[  10.885591] Audio.DSP <Error> audio_core/hle/source.cpp:DequeueBuffer:259: Looped buffers are unimplemented at the moment
[  12.391301] Service.FS <Warning> core/hle/service/fs/archive.cpp:HandleSyncRequest:175: (STUBBED) File command OpenLinkFile Path: [Binary: 000000000000000000000000]
[  12.752502] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2b (0xac000000)
[  12.752524] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2a (0xa8000000)
[  12.752531] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2b (0xad000000)
[  12.752535] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2a (0xa8000000)
[  12.752544] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2b (0xae800000)
[  12.752548] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2a (0xa8000000)
[  12.752554] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2b (0xacc00000)
[  12.752558] HW.GPU <Critical> video_core/shader/shader_jit_x64.cpp:Compile_NextInstr:814: Unhandled instruction: 0x2a (0xa8000000)
```

**Comment by yuriks** [comment]
Looks like missing geoshaders.

**Comment by joaozin23** [comment]
![esboco 2](https://cloud.githubusercontent.com/assets/17303493/21295949/1733c258-c547-11e6-9c31-4c6737eb78e7.png)
bleeding-edge-102

**Comment by joaozin23** [comment]
new super mario bros 2 3d land

**Comment by yuriks** [comment]
@joaozin23 Please stop posting off-topic on unrelated issues.

**Comment by wwylele** [comment]
Fixed by #2865.
